pkg/rke2/machine_filters.go

```diff
@@ -76,7 +76,7 @@ func matchesRKE2BootstrapConfig(machineConfigs map[string]*bootstrapv1.RKE2Confi
 			cmds := []string{}
 
 			for _, cmd := range machineConfig.Spec.PostRKE2Commands { // Filter out commands that are injected by the Rancher Turtles webhook
-				if cmd == "sudo sh /opt/system-agent-install.sh" {
+				if cmd == "sh /opt/system-agent-install.sh" {
```

Comment:
before merging this, can we figure out why system-agent install script requires root permissions to run https://github.com/rancher/system-agent/blob/main/install.sh#L886?

Reply:
To add and enable systemd unit with a binary running as root you would need high privileges. Bootstrap commands already at root, so this works.